File: Data Structures & Algorithms/two-integer-sum/submission-5.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        i,j=0,1
        while i < len(nums):
            x = nums[i]
            while j < len(nums):
                y = nums[j]
                if x + y == target:
                    indexs = (i,j)
                    return list(indexs)
                j = j +1
            i = i +1
            j = i+1
        return []

File: Data Structures & Algorithms/find-duplicate-integer/submission-5.java
class Solution {
    public int findDuplicate(int[] nums) {
        var set = new HashSet<Integer>();
        var slow = 0;
        var fast = 0;
        while (true) {
            slow = nums[slow];
            fast = nums[nums[fast]];
            if (slow == fast)
                break;
        }
        var start = 0;
        // while slow != fast
        while (true) {
            start = nums[start];
            slow = nums[slow];
            if (start == slow) {
                return slow;
            }
        }
    }
}
